benchmarks/transactions-generator/src/lib.rs

```
@@ -172,53 +170,80 @@ impl TxGenerator {
     }
 
     fn start_transactions_loop(
-        params: &TxGeneratorConfig,
+        config: &TxGeneratorConfig,
         client_sender: ClientSender,
         view_client_sender: ViewClientSender,
         runner_state: RunnerState,
-    ) -> anyhow::Result<task::JoinHandle<()>> {
-        let mut tx_interval = tokio::time::interval(Duration::from_micros(
-            1_000_000 * params.thread_count / params.tps,
-        ));
+    ) -> anyhow::Result<Vec<task::JoinHandle<()>>> {
         // TODO(slavas): generate accounts on the fly?
-        let mut accounts = account::accounts_from_dir(&params.accounts_path)?;
+        let mut accounts = account::accounts_from_dir(&config.accounts_path)?;
         if accounts.is_empty() {
             anyhow::bail!("No active accounts available");
         }
 
-        Ok(tokio::spawn(async move {
-            let mut rnd: StdRng = SeedableRng::from_entropy();
+        let sender = view_client_sender.clone();
+        let accounts = Arc::new(
+            std::thread::spawn(move || {
+                let rt = tokio::runtime::Runtime::new().unwrap();
+                rt.block_on(async move {
+                    for account in accounts.iter_mut() {
+                        let (id, pk) = (account.id.clone(), account.public_key.clone());
+                        if let Ok(nonce) = Self::get_client_nonce(sender.clone(), id, pk).await {
```

should we "print" errors if this fails instead of silently dropping?

---

Always a good idea. Fixed.